File: problem/Array_Elements_Zero/countValidSelections.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var countValidSelections = function(nums) {
    let total = nums.reduce((a, b) => a + b, 0);
    let ls = 0, cases = 0;
    for (let num of nums) {
        let rs = total - ls - num;
        if (num === 0) {
            if (ls === rs) {
                cases += 2;
            } else if (ls === rs - 1 || ls - 1 === rs) {
                cases++;
            }
        }
        ls += num;
    }
    return cases;
};

File: problem/Array_Elements_Zero/countValidSelections.c
int countValidSelections(int *nums, int numsSize)
{
    int total = 0;
    for (int i = 0; i < numsSize; i++)
    {
        total += nums[i];
    }
    int ls = 0, cases = 0;
    for (int i = 0; i < numsSize; i++)
    {
        int rs = total - ls - nums[i];
        if (nums[i] == 0)
        {
            if (ls == rs)
            {
                cases += 2;
            }
            else if (ls == rs - 1 || ls - 1 == rs)
            {
                cases++;
            }
        }
        ls += nums[i];
    }
    return cases;
}